cellmlmanip/model.py

```diff
@@ -1048,4 +1106,9 @@ def __str__(self):
     @property
     def rdf_identity(self):
         """The RDF identity for this variable, or ``None`` if no cmeta id."""
-        return create_rdf_node('#' + self.cmeta_id) if self.cmeta_id else None
+        if self._cmeta_id is None:
+            return None
+        if self._rdf_identity is None:
+            self._rdf_identity = create_rdf_node('#' + self._cmeta_id)
+        return self._rdf_identity
```

Comment:
Because this is now cached, if a user has accessed `rdf_identity` then transfers the cmeta_id away, *both* variables will end up with the same `rdf_identity` cached.

I think it'd be safer to add a `_set_cmeta_id()` method on `VariableDummy` that sets both `self._cmeta_id` and `self.rdf_identity`, and remove this `@property` method.

Reply:
:flushed: woops!